docs/python-development.rst

```diff
@@ -1,7 +1,14 @@
-
 Python development
 ==================
 
+Release process
+---------------
+
+Prior to release a new version, the maintainer:
+ - can create a GitHub project if required listing the PRs to be considered in the upcoming release (optional)
```

Comment:
Is there a past or sample project we could link to as a template?

Reply:
We have used some but they were not made public.
We could try to create a public one for the next body of work.

Reply:
:+1: